x-pack/platform/plugins/shared/actions/server/application/connector/methods/get_axios_instance/get_axios_instance.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AxiosInstance } from 'axios';
+import type { RawAction } from '../../../../types';
+import { getActionKibanaPrivileges } from '../../../../lib/get_action_kibana_privileges';
+import { isPreconfigured } from '../../../../lib/is_preconfigured';
+import { isSystemAction } from '../../../../lib/is_system_action';
+import { ACTION_SAVED_OBJECT_TYPE } from '../../../../constants/saved_objects';
+import type { ActionsClientContext } from '../../../../actions_client';
+import { validateSecrets } from '../../../../lib';
+
+type ValidatedSecrets = Record<string, unknown>;
+
+export type GetAxiosInstanceWithAuthFn = (secrets: ValidatedSecrets) => Promise<AxiosInstance>;
+
+export async function getAxiosInstance(
+  context: ActionsClientContext,
+  actionId: string
+): Promise<AxiosInstance> {
+  const {
+    actionTypeRegistry,
+    authorization,
+    encryptedSavedObjectsClient,
+    getAxiosInstanceWithAuth,
+    inMemoryConnectors,
+    isESOCanEncrypt,
+    logger: log,
+    request,
+    spaces,
+    unsecuredSavedObjectsClient,
+  } = context;
+
+  let actionTypeId: string | undefined;
+
+  try {
+    if (isPreconfigured(context, actionId) || isSystemAction(context, actionId)) {
+      const connector = inMemoryConnectors.find(
+        (inMemoryConnector) => inMemoryConnector.id === actionId
+      );
+
+      actionTypeId = connector?.actionTypeId;
+    } else {
+      const { attributes } = await unsecuredSavedObjectsClient.get<RawAction>(
+        ACTION_SAVED_OBJECT_TYPE,
+        actionId
+      );
+
+      actionTypeId = attributes.actionTypeId;
+    }
+  } catch (err) {
+    log.debug(`Failed to retrieve actionTypeId for action [${actionId}]`, err);
+    throw err;
+  }
+
+  await authorization.ensureAuthorized({
+    operation: 'execute',
+    additionalPrivileges: getActionKibanaPrivileges(context, actionTypeId),
```

Comment:
is this necessary? i thought the additional privileges depended on just the subaction parameters. maybe we can skip calling this function and just pass in `[]`